content/blog/2021/Bitcoin_Core_RPC_Demo.md

```diff
@@ -0,0 +1,666 @@
+---
+title: "Using BDK to build a wallet backed by a Bitcoin Core full node"
+description: "Tutorial showing usage of Bitcoin core backend with BDK wallet"
+authors:
+    - Rajarshi Maitra
+date: "2021-08-21"
+tags: ["tutorial", "BDK", "Bitcoin Core", "RPC", "Wallet"]
+hidden: true
+draft: false
+---
+
+### Introduction
+BDK wallet developer library can be used to easily deploy wallets with various kind of blockchain backend suppot, like `electrum`, `explora`, `compact-filters` ([BIP157](https://github.com/bitcoin/bips/blob/master/bip-0157.mediawiki)) etc. With the latest release of BDK [`v0.10.0`](https://github.com/bitcoindevkit/bdk/tree/0.10.0), BDK now supports Bitcoin Core as a blockchain backend. BDK talks with Bitcoin Core using rust-bitcoin's [bitcoincore-rpc](https://github.com/rust-bitcoin/rust-bitcoincore-rpc) library.
+
+This allows wallet devs to quickly deploy their wallet that can talk to a bitcoin full node (home raspi nodes) out of the box. Wallet devs don't need to worry about connecting to a full node with correct RPC calls, all of that is handled by BDK under the hood. All they need is to identify the full node's RPC IP address and the correct RPC credentials.
+
+In this tutorial we will see how to write a very simplistic wallet code that can connect to a bitcoin core node and maintain its balance and make transactions.
+
+Unlike other tutorials, we will not use `bdk-cli` tools, but instead write rust code directly using `BDK` devkit. In the end we will end up with our own simple bitcoin wallet.
+
+### Prerequisite
+To run with this tutorial you would need to have a bitcoin core node running in regtest mode. Get the bitcoin core binary either from the [bitcoin core repo](https://bitcoincore.org/bin/bitcoin-core-0.21.1/) or [build from source](https://github.com/bitcoin/bitcoin/blob/v0.21.1/doc/build-unix.md). Alternatively you can use the `bitcoin-regtest-box` docker file maintained by BDK project [here](https://github.com/bitcoindevkit/bitcoin-regtest-box).
```

Comment:
It looks like the `bitcoin-regtest-box` has been updated to support cookie auth now. I had to use the docker images before this [commit](0db538115ef8e954a97c31e6c7e6f63b16fd2bb2) to get it working. 

What can we do to make this a bit clearer?

Reply:
Ah ok, I need to update the tutorial to handle cookie auth i think..

Reply:
I'd like archive `bitcoin-regtest-box` and update our docs/blogs to not use it. As a start I've created https://github.com/bitcoindevkit/bitcoin-regtest-box/pull/12 with alternative instructions for local regtest testing.

Reply:
Opted for using Polar which is basically using Docker under the hood. We could probably remove mentioning `bitcoin-regtest-box` as it's no longer maintained. Some docs around setting up a regtest bitcoind locally would be handy but that could be separate from this tutorial IMO.

Reply:
I actually use my own local regtest node instead of bitcoin-regtest-box. Do you think we should add local regtest setup somewhere? Or shift to Polar?